leases/utils.py

```diff
@@ -43,6 +45,52 @@ def calculate_season_end_date(lease_end: date = None) -> date:
     return default
 
 
+def calculate_winter_season_start_date(lease_start: date = None) -> date:
+    """Return the date when the winter season starts
+
+    If the current date is after the season end date,
+    it returns next year's default start date.
+
+    Winter Storage Leases always start on 15.9 the earliest
+    """
+    today = date.today()
+    default = date(day=15, month=9, year=today.year)
+
+    if lease_start:
+        return default.replace(year=lease_start.year)
```

Comment:
this is a bit tricky, since WS lease can start between 1.1 and 10.6, in which case we should do `default.replace(year=lease_start.year - 1)`

Reply:
:hm: right, it's not as straightforward as the summer season one.